servicenow-case-to-salesforce-case/main.bal

```diff
@@ -0,0 +1,80 @@
+import ballerina/http;
+import ballerina/io;
+import ballerina/log;
+import ballerina/mime;
+import ballerina/time;
+import ballerina/url;
+import ballerinax/salesforce as sf;
+
+configurable string servicenowInstance = ?;
+configurable string syncData = ?;
+configurable string serviceNowUsername = ?;
+configurable string serviceNowPassword = ?;
+configurable sf:ConnectionConfig salesforceConfig = ?;
+
+public function main() returns error? {
+    DateRange|error fetchPeriod = calculateFetchingPeriod();
+    if fetchPeriod is error {
+        log:printError("Error while calculating fetching period: ", fetchPeriod);
+        return;
+    }
+
+    CaseData[]|error cases = fetchCasesFromServiceNow(fetchPeriod.'start, fetchPeriod.end);
+    if cases is error {
+        log:printError("Error while fetching cases from ServiceNow: ", cases);
+        return;
+    }
+
+    error? e = addCasesToSalesforce(cases);
+    if e is error {
+        log:printError("Error while adding cases to Salesforce: ", e);
+        return;
+    }
+
+    check io:fileWriteString(syncData, check time:civilToString(fetchPeriod.now));
+    log:printInfo("Updated salesforce with cases from servicenow." +
+        " Cases added: " + cases.length().toString() +
+        " Update Timestamp: " + check time:civilToString(fetchPeriod.now));
+}
+
+function calculateFetchingPeriod() returns DateRange|error {
+    string lastFetchString = check io:fileReadString(syncData);
+    time:Civil lastFetch = check time:civilFromString(lastFetchString);
+    string 'start = string `'${lastFetch.year}-${lastFetch.month}-${lastFetch.day}','${lastFetch.hour}:${lastFetch.minute}:00'`;
+    time:Civil now = time:utcToCivil(time:utcNow());
+    string end = string `'${now.year}-${now.month}-${now.day}','${now.hour}:${now.minute}:00'`;
+    return {'start, end, now};
+}
+
+function fetchCasesFromServiceNow(string fetchFrom, string fetchTill) returns CaseData[]|error {
+    string query = string `sys_created_onBETWEENjavascript:gs.dateGenerate(${fetchFrom})@javascript:gs.dateGenerate(${fetchTill})`;
```

Comment:
Can we have the following order for the statements? The reason is no point of creating `query` if the HTTP client initialization failed.

```ballerina
http:Client servicenow = check new (string `https://${servicenowInstance}.service-now.com/api/sn_customerservice`);
    string serviceNowCredentials = check mime:base64Encode(serviceNowUsername + ":" + serviceNowPassword, "UTF-8").ensureType();
    string query = string `sys_created_onBETWEENjavascript:gs.dateGenerate(${fetchFrom})@javascript:gs.dateGenerate(${fetchTill})`;
```